mantidproject/mantid
# Create script for deleting unused documentation images

**Description of work**
This PR creates a script in the `tools` directory which can be used for pruning documentation images. It will prune any images which are not referenced somewhere else in the Mantid repo. The assumption it makes is that if an image is used, the name of the image will be mentioned somewhere in a documentation file.

Usage:
```sh
./tools/prune_images.sh --dry-run
```

**To test:**
Check that this build package from branch has passed:
https://builds.mantidproject.org/job/build_packages_from_branch/592/

You can try run the script locally **on main**, with and without the `dry-run` flag:
```sh
./tools/prune_images.sh --dry-run
```

Check if I have missed any image file types that we may want to delete. Currently this only deletes `*.png`, `*.PNG` and `*.jpeg`

*There is no associated issue.*

*This does not require release notes* because **it is a change not visible to users**

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
